### PR TITLE
Support for 128-bits secure multiparty computing

### PR DESCRIPTION
**Pull request by joyoFeng**
Support for 128-bits secure multiparty computing, providing higher precision floating-point computing capability.

### CODE REVIEW

File: cc/modules/protocol/mpc/snn/src/impl/mpc_basic.cpp [hunk 1]
```diff
@@ -20,6 +20,8 @@
 namespace rosetta {
 namespace snn {
 
+#define PRSS_OPT_VALUE 2
```
Comment (vingstar) [inline cc/modules/protocol/mpc/snn/src/impl/mpc_basic.cpp:23]: Hi, could you describe the meaning of this Macro?

File: cc/modules/protocol/mpc/snn/src/impl/mpc_basic.cpp [hunk 2]
```diff
@@ -81,9 +82,9 @@ int ComputeMSB::funcComputeMSB3PC(const vector<mpc_t>& a, vector<mpc_t>& b, size
     populateRandomVector<mpc_t>(r2, size, "a_2", "POSITIVE");
     for (size_t i = 0; i < size; ++i) {
       if (r1[i] == MINUS_ONE)
-        r1[i]--;
+        r1[i] = 0;////[huanggaofeng] fix for odd ring Z_{2^128 - 1}, origin: r1[i]--
```
Comment (vingstar) [inline cc/modules/protocol/mpc/snn/src/impl/mpc_basic.cpp:85]: Could you explain why this works?

File: python/latticex/rosetta/__init__.py [hunk 3]
```diff
@@ -38,6 +38,10 @@ def __getattr__(self, attr):
 sys.stdout = ZeroBufferOut(sys.stdout)
 
 
+if 'ROSETTA_MPC_128' in os.environ and os.environ['ROSETTA_MPC_128'] == 'ON':
```
Comment (vingstar) [inline python/latticex/rosetta/__init__.py:41]: I see the same codes in multi files, could you make them in a separate file and import it in other files?

File: cc/modules/protocol/mpc/snn/src/impl/mpc_basic.cpp (hunk 1 above)
Comment (joyoFeng) [reply cc/modules/protocol/mpc/snn/src/impl/mpc_basic.cpp:23]: This `PRSS_OPT_VALUE` means the optimization for communication with PRSS, for history reason and to make it readable, I keep the origin code and use `PRSS_OPT_VALUE`  to orgnize them.

File: cc/modules/protocol/mpc/snn/src/impl/mpc_basic.cpp (hunk 2 above)
Comment (joyoFeng) [reply cc/modules/protocol/mpc/snn/src/impl/mpc_basic.cpp:85]: In odd ring `Z_{L -1}`,  -1 means 0, which is -1 in `Z_L`

**Comment by vingstar** [comment]
LGTM, I will merge it.

File: python/latticex/rosetta/__init__.py (hunk 3 above)
Comment (joyoFeng) [reply python/latticex/rosetta/__init__.py:41]: I have push request to fix the issue: [https://github.com/LatticeX-Foundation/Rosetta/pull/31/commits/485acf4a267f633d581d9a27edd9e3abd81a9117](update)

**Pull request merged**